examples/test_memory_leaks.ml

```diff
@@ -12,7 +12,7 @@ let test1 size =
 let test2 size =
   let h = init () in
   let s = String.make size 'a' in
-  set_mimepost h [{encoding=CURLMIME_BINARY;headers=[];subparts=[];data=CURLMIME_DATA s}];
+  set_mimepost h [{encoding=CURLMIME_BINARY;headers=[];subparts=[];name=None;data=CURLMIME_DATA s}];
```

Comment:
we can avoid breaking existing code by adding new ctor `CURLMIME_DATA_WITH_NAME(S) of data` with `type data = { data : string; name : string option; filename : string option }`

Reply:
My impression is that it is possible to add a name and/or filename to a part filled with either `curl_mime_data` or `curl_mime_filedata`.  Shall I add two constructors: `CURLMIME_DATA_WITH_NAME` and `CURLMIME_FILEDATA_WITH_NAME`?

Reply:
right. Then `{data : data_source; ...} with `data_source = String of string | File of string` ?

Reply:
I made the changes.